sitemedia/js/components/SearchFacet.js

```diff
@@ -0,0 +1,99 @@
+import { mapActions } from 'vuex'
+import RangeFacet from './RangeFacet'
+
+export default Vue.component('SearchFacet', {
+    template: `
+    <sui-grid-column :width="width" class="field">
+        <label :for="label">{{ label }}</label>
+        <template v-if="type === 'text'">
+            <sui-input v-if="search" iconPosition="left" icon="search" placeholder="Search or select" v-model="filter" />
+            <div v-if="search" class="rolodex">
+                <button
+                    v-for="letter in alphabet"
+                    :key="letter"
+                    class="letter"
+                    :class="{active: alphaFilter == letter}"
+                    @click.prevent="alphaFilter = alphaFilter == letter ? '' : letter"
+                >
+                {{ letter }}
+                </button>
+            </div>
+            <div class="facets">
+                <label v-if="availableChoices.length == 0">No results</label>
+                <div
+                    class="ui checkbox"
+                    v-for="choice of choices"
+                    v-show="availableChoices.includes(choice.value)"
+                    :key="choice.value"
+                >
+                    <input
+                        type="checkbox"
+                        @input="toggleFacetChoice(choice)"
+                        :name="choice.facet"
+                        :value="choice.value"
+                        :checked="choice.active"
+                    >
+                    <label>{{ choice.value }} <span class="count">{{ choice.count }}</span></label>
+                </div>
+            </div>
+        </template>
+        <range-facet v-if="type === 'range'" :label="label" :name="name" :width="width" :choices="choices">
+        </range-facet>
+    </sui-grid-column>
+    `,
+    components: {
+        RangeFacet
+    },
+    data() {
+        return {
+            filter: '',
+            alphaFilter: '',
+            alphabet: String.fromCharCode(...Array(91).keys()).slice(65)
```

Comment:
I think you need a bit more here to determine which letters are actually present in the data so you can disable the letter choices that aren't.

Reply:
Maybe this would be good to extrapolate away as a Vuex getter that takes an alphabetical facet's results and returns an alphabet with letters not present marked (or vice-versa)?